internal/build/hooks.go

```diff
@@ -0,0 +1,60 @@
+package build
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"sigs.k8s.io/yaml"
+
+	"github.com/shipa-corp/ketch/internal/errors"
+)
+
+// Shipa holds the decoded results of a shipa.yaml file
+type Shipa struct {
+	Hooks *Hooks `json:"hooks,omitempty"`
+}
+
+// Hooks contain build hooks from a shipa yaml file. Hooks are shell scripts that
+// run during deployments. See https://learn.shipa.io/docs/shipayml
+type Hooks struct {
+	Build []string `json:"build,omitempty"`
+}
+
+func getHooks(workingDir string, shipaFileSearchPaths []string) ([]string, error) {
```

Comment:
I think that we can validate and decode `ketch.yaml` at the begging of `appDeploy` function and then pass it to `GetSourceHandler` and `appDeployImage`